contracts/farms/RevoMobiusFarmBot.sol

```diff
@@ -0,0 +1,209 @@
+        require(_lpAmount > 0, "Cannot invest in farm because lpAmount is 0");
+        stakingToken.safeIncreaseAllowance(address(liquidityGauge), _lpAmount);
+        liquidityGauge.deposit(_lpAmount);
+    }
+
+    function _withdraw(uint256 _lpAmount) internal override {
+        liquidityGauge.withdraw(_lpAmount);
+    }
+
+    function _claimRewards() internal whenNotPaused {
+        minter.mint(address(liquidityGauge));
+        liquidityGauge.claim_rewards(address(this));
+    }
+
+    function compound(
+        address[][] memory _paths,
+        uint256[] memory _minAmountsOut,
+        uint256 _minSwapOut,
+        uint256 _minLiquidity,
+        uint256 _deadline
+    ) external ensure(_deadline) onlyRole(COMPOUNDER_ROLE) whenNotPaused {
+        // Important safety checks
+        require(
+            _paths.length == rewardsTokens.length,
+            "Parameter _paths must have length equal to rewardsTokens"
+        );
+        require(
+            _minAmountsOut.length == rewardsTokens.length,
+            "Parameter _minAmountsOut must have length equal to rewardsTokens"
+        );
+        for (uint256 i = 0; i < _paths.length; i++) {
+            require(
+                _paths[i][_paths[i].length - 1] ==
+                    address(celoNativeStakingToken),
+                "Each swap path must end with the Celo-native staking token"
+            );
+        }
+
+        // Get native and external rewards
+        _claimRewards();
+
+        // Get the current balance of rewards tokens
+        uint256[] memory _tokenBalances = new uint256[](rewardsTokens.length);
+        for (uint256 i = 0; i < rewardsTokens.length; i++) {
+            _tokenBalances[i] = rewardsTokens[i].balanceOf(address(this));
+        }
+
+        // Swap rewards for the Celo-native LP constituent token
+        for (uint256 i = 0; i < rewardsTokens.length; i++) {
+            UniswapRouter.swap(
+                router,
+                _paths[i],
+                _tokenBalances[i],
+                rewardsTokens[i],
+                _minAmountsOut[i],
+                _deadline
+            );
+        }
+
+        uint256 halfCeloNativeStakingTokenBalance = celoNativeStakingToken
+            .balanceOf(address(this)) / 2;
+
+        // Take half the Celo-native staking token and swap it for the bridged version
+        celoNativeStakingToken.safeIncreaseAllowance(
+            address(swap),
+            halfCeloNativeStakingTokenBalance
+        );
+
+        swap.swap(
+            celoNativeStakingTokenIndex,
+            bridgedStakingTokenIndex,
+            halfCeloNativeStakingTokenBalance,
+            _minSwapOut,
+            _deadline
+        );
+
+        uint256 stakingToken0Balance = stakingToken0.balanceOf(address(this));
+        uint256 stakingToken1Balance = stakingToken1.balanceOf(address(this));
+
+        // Make sure the dynamic `amounts` variable is empty, then add the balances and mint liquidity
```

Comment:
This is really, really weird IMO. I'm not sure how to deal with this. The `addLiquidity` signature specifies that `amounts` lives in calldata as a dynamically sized array, which would be fine if the compounder was passing `amounts` in manually. A fixed-size `memory` array prevents the contract from compiling; this was the only workaround I could come up with..

We could also try changing the `calldata` argument to `memory` in the `ISwap` interface and see what happens, but that feels... wrong?

Reply:
another lame idea would be to have the compounding take place in sort of a staggered way, where the LP is minted from the staking token balance left over from the last time compound was called. That way, the compounder could know in advance what the amounts would be. 

I'm not sure that's better, though. Just weird in a different way, really.